pg_backup_api/pg_backup_api/server_operation.py

```diff
@@ -74,42 +177,106 @@ def copy_and_validate_options(self, general_options):
 
         return job_data
 
-    def write_jobs_files(self, general_options):
+    def write_jobs_files(self, general_options: Dict[str, str]) -> None:
+        """
+        Create a job file to represent a requested recovery operation.
+
+        File is created under :attr:`jobs_basedir`. Its content is gotten
+        through :meth:`copy_and_validate_options`.
+
+        .. note::
+            If the directories pointed by :attr:`jobs_basedir` and
+            :attr:`output_basedir` do not exist yet, this method will take care
+            of creating them.
+
+        :param general_options: a Python :class:`dict` containing the values
+            for the keys of :data:`REQUIRED_OPTIONS`.
+        """
         for directory in (self.jobs_basedir, self.output_basedir):
             if not os.path.exists(directory):
                 os.makedirs(directory)
 
         job_data = self.copy_and_validate_options(general_options)
         self.__create_file(self.jobs_basedir, job_data)
 
-    def create_output_file(self, content_file):
+    def create_output_file(self, content_file: Dict[str, str]) -> None:
+        """
+        Create an output file to represent the output of a recovery operation.
+
+        :param content_file: the content to be written in the created file.
+            Expects a Python dictionary which will be converted to JSON.
+        """
         self.__create_file(self.output_basedir, content_file)
 
-    def __create_file(self, file_type, content):
+    def __create_file(self, parent_dir: str, content: Dict[str, str]) -> None:
+        """
+        Create a file under *parent_dir* with *content*.
+
+        The file will be named ":attr:`operation_id`.json".
+
+        :param parent_dir: directory under which the file should be created.
+        :param content: content to be written into the file. Expects a Python
+            dictionary which will be converted to JSON.
+
+        :raises:
+            :exc:`Exception` if :attr:`operation_id` has not been filled before
+                calling this method, or if the file already exists.
+        """
         if not self.operation_id:
             raise Exception("operation_id is required here")
 
-        file_name = "{}.json".format(self.operation_id)
-        fpath = join(file_type, file_name)
+        file_name = f"{self.operation_id}.json".
```

Comment:
Are you missing something after the trailing dot?
Said differently, is that dot a leftover from the original code?

Reply:
I guess I added that by mistake after running `flake8`, otherwise it would have showed something like this when I ran the linter:

```
pg_backup_api/server_operation.py:228:50: E999 SyntaxError: invalid syntax
```